libmscore/textlinebase.cpp

```diff
@@ -106,11 +106,34 @@ void TextLineBaseSegment::draw(QPainter* painter) const
             textlineLineWidth *= mag();
       QPen pen(color, textlineLineWidth, tl->lineStyle());
       QPen solidPen(color, textlineLineWidth, Qt::SolidLine);
-      if (tl->lineStyle() == Qt::CustomDashLine) {
-            QVector<qreal> dashes { tl->dashLineLen(), tl->dashGapLen() };
-            pen.setDashPattern(dashes);
+
+      //Replace generic Qt dash patterns with improved equivalents to show true dots
+      QVector<qreal> dotted        = { 0.01, 1.99 }; // tighter than Qt Dotline equivalent { 0.01, 2.99 }
+      QVector<qreal> dashed        = { 3.0, 3.0 };
+      QVector<qreal> dashDotted    = { 3.01, 2.99, 0.01, 2.99 };
+      QVector<qreal> dashDotDotted = { 3.01, 2.99, 0.01, 2.99, 0.01, 2.99 };
+      QVector<qreal> customDashes  = { tl->dashLineLen(), tl->dashGapLen() };
```

Comment:
Here too, add one line width?

Reply:
Did better, { 3.0, 3.0 }  is a dash with its gap (compensated already from {4,2}) and {0.01, 2.99} is a dot with its gap.

Reply:
{4.0, 2.0} is DashLine default. Compensated to {3.0, 3.0} for caps.
{1.0, 2.0} is DotLine default. Compensated for caps would be {0.0, 3.0} but zero doesn't work so I give 1 cent back from the gap to the dash - it would be { 0.1, 2.99} but our version of DotLine has smaller gaps { 0.01, 1.99}
The dashDotted and dashDotDotted use nominally 3.0 on every gaps. The 2.99 compensation for 0.01 dashes (ie. dots) is to keep the most consistent output along the line between dash/dashdot etc. so the gaps look aligned between styles when side to side.
Not that someone would tightly compare lines, but it looks neat if the composer wants to add instructions about lines somewhere.
![image](https://user-images.githubusercontent.com/2843953/88081564-ece0c180-cb56-11ea-9d9b-d7b52f39155e.png)

Reply:
We can't have real dots for the dashDotted and dashDotDotted lines, can we?

Reply:
Only if we assume RoundCaps for the dashes on these two, leaving only dashes with the SquareCaps. 
Ideally, a new PR could add this Cap Style as an option in inspector / styles / etc... A new property to save to files and deal with. 
Would that be overkill?

Reply:
In any case, mixing straight dashes with round dots in the same line is not possible with Qt styles. Only if we create a method to calculate dashes and dots separately and then draws two overlapping lines. Just for this...

Reply:
Actually I had a simple idea to draw a line with mixed round dots and sharp dashes. Would it be worth it?
I'll give it a try.